server/game/core/ability/abilityTargets/CardTargetResolver.ts

```diff
@@ -58,6 +58,15 @@ export class CardTargetResolver extends TargetResolver<ICardTargetResolver<Abili
     }
 
     protected override resolveInner(context: AbilityContext, targetResults, passPrompt, player: Player) {
+        // a player can always choose not to pick a card from a zone that is hidden from their opponents (SWU Comp Rules 2.0 1.17.4)
+        // TODO: test if picking a card from an opponent's usually hidden zone(e.g. opponent's hand) works as expected(the if block here should be skipped)
+        if (CardTargetResolver.allZonesAreHidden(this.properties.locationFilter, typeof this.properties.controller === 'function' ? this.properties.controller(context) : this.properties.controller)) {
```

Comment:
Hey unfortunately I realized there's a slight bug here - this condition only applies if there is some restriction on the card being drawn. The rule states that "a player may choose to act as if they have fewer options than they do," but that means if the ability lets you draw _any_ card, you are not revealing hidden information by choosing and you must do it.

That will probably be a slightly bigger work item to distinguish the two cases so I'm fine if for now you just want to put TODOs here and in SelectCardSystem

Reply:
Would it be too simple to just check if `properties` has a `cardTypeFilter` or `cardCondition`?

Reply:
Yeah that was my thought, would just have to add some computed property on CardSelector like `get hasAnyFilter() { return !!this.cardCondition && !!this.cardTypeFilter; }` or something. I just know this PR review has gone long already so don't want to push you to do more in this one if you'd rather merge here and save it for later.